packages/grafana-ui/src/components/Button/ToolbarButton.tsx

```diff
@@ -19,27 +20,31 @@ export interface Props extends HTMLAttributes<HTMLButtonElement> {
   fullWidth?: boolean;
   /** reduces padding to xs */
   narrow?: boolean;
+  /** variant */
+  variant?: ButtonVariant;
 }
 
 export const ToolbarButton = forwardRef<HTMLButtonElement, Props>(
-  ({ tooltip, icon, className, children, imgSrc, fullWidth, isOpen, narrow, ...rest }, ref) => {
+  ({ tooltip, icon, className, children, imgSrc, fullWidth, isOpen, narrow, variant, ...rest }, ref) => {
     const styles = useStyles(getStyles);
 
-    const contentStyles = cx({
-      [styles.content]: true,
-      [styles.contentWithIcon]: !!icon,
-      [styles.contentWithRightIcon]: isOpen !== undefined,
-    });
-
     const buttonStyles = cx(
       {
         [styles.button]: true,
         [styles.buttonFullWidth]: fullWidth,
         [styles.narrow]: narrow,
+        [styles.primaryVariant]: variant === 'primary',
+        [styles.destructiveVariant]: variant === 'destructive',
       },
       className
     );
```

Comment:
If the styles are renamed to 'primary' and 'destructive', we'd then simplified this code a bit, which is also easier to maintain in case more variants are added. 
```suggestion
    const buttonStyles = cx(
      {
        [styles.button]: true,
        [styles.buttonFullWidth]: fullWidth,
        [styles.narrow]: narrow,
      },
      styles[variant]
      className
    );
```

Reply:
ah good idea!

Reply:
except when variant is undefined

Reply:
could not make this suggestion work, without introducing messy logic around undefined